aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/OutboxMutationFailedEvent.java

```diff
@@ -55,46 +55,31 @@ private OutboxMutationFailedEvent(
      * @param errors the list of graphQL errors that caused the failure
      * @param <M> Class type of the model.
      * @return Outbox mutation error event.
-     * @throws DataStoreException if unexpected mutation type
      */
     @NonNull
     public static <M extends Model> OutboxMutationFailedEvent<M> create(
             @NonNull PendingMutation<M> pendingMutation,
             @NonNull List<GraphQLResponse.Error> errors
-    ) throws DataStoreException {
+    ) {
         Objects.requireNonNull(pendingMutation);
         Objects.requireNonNull(errors);
-        final MutationType opType;
-        switch (pendingMutation.getMutationType()) {
-            case CREATE:
-                opType = MutationType.CREATE;
-                break;
-            case UPDATE:
-                opType = MutationType.UPDATE;
-                break;
-            case DELETE:
-                opType = MutationType.DELETE;
-                break;
-            default:
-                throw new DataStoreException(
-                    "Invalid operation type.",
-                    AmplifyException.REPORT_BUG_TO_AWS_SUGGESTION
-                );
-        }
-        return new OutboxMutationFailedEvent<>(errors, opType, pendingMutation.getMutatedItem());
+        MutationErrorType errorType = MutationErrorType.fromGraphQLErrors(errors);
+        String operation = pendingMutation.getMutationType().name();
+        MutationType opType = MutationType.valueOf(operation);
```

Comment:
Hm, I'm confused by this logic. You're converting an _error type_ (`UNAUTHORIZED`) to a _`MutationType`_ (`UPDATE`, etc.)?

Reply:
It is merely converting it from `PendingMutation.Type` internal enum to a customer-exposed `MutationType` enum. However, I realized that this enum comes from `aws-api-appsync` module, which is a problem on its own since we want to abstract away GraphQL from here. I will create another enum type within datastore module.

Reply:
Addressed in [this commit](https://github.com/aws-amplify/amplify-android/pull/976/commits/01ee3ee914060394f95ac602fa3b84a3c4284fc3) (Needs review).